engine/classes/Elgg/Di/ServiceProvider.php

```diff
@@ -204,8 +204,16 @@ public function __construct(\Elgg\AutoloadManager $autoload_manager) {
 			}
 
 			$handler = new \Elgg\Http\DatabaseSessionHandler($c->db);
-			$storage = new \Elgg\Http\NativeSessionStorage($params, $handler);
-			return new \ElggSession($storage);
+
+			// session.cache_limiter is unfortunately set to "" by the NativeSessionStorage constructor,
+			// so we must capture and inject it directly.
+			$options = [
+				'cache_limiter' => session_cache_limiter(),
+			];
+			$storage = new \Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage($options, $handler);
```

Comment:
Yikes. Can we add a `use` statement at the top for these beasts? Maybe just for `HttpFoundation` or `Session as SymfonySession` or something?